TOOLS/lua/autocrop.lua

```diff
@@ -1,114 +1,258 @@
--- This script uses the lavfi cropdetect filter to automatically
--- insert a crop filter with appropriate parameters for the currently
--- playing video.
---
--- It registers the key-binding "C" (shift+c), which when pressed,
--- inserts the filter vf=lavfi=cropdetect. After 1 second, it then
--- inserts the filter vf=crop=w:h:x:y, where w,h,x,y are determined
--- from the vf-metadata gathered by cropdetect. The cropdetect filter
--- is removed immediately after the crop filter is inserted as it is
--- no longer needed.
---
--- If the "C" key is pressed again, the crop filter is removed
--- restoring playback to its original state.
---
--- Since the crop parameters are determined from the 1 second of video
--- between inserting the cropdetect and crop filters, the "C" key
--- should be pressed at a position in the video where the crop region
--- is unambiguous (i.e., not a black frame, black background title
--- card, or dark scene).
---
--- The default delay between insertion of the cropdetect and
--- crop filters may be overridden by adding
---
--- --script-opts=autocrop.detect_seconds=<number of seconds>
---
--- to mpv's arguments. This may be desirable to allow cropdetect more
--- time to collect data.
+--[[
+This script uses the lavfi cropdetect filter to automatically
+insert a crop filter with appropriate parameters for the currently
+playing video.
+
+It registers the key-binding "C" (shift+c), which when pressed,
+inserts the filter vf=lavfi=cropdetect. After 1 second, it then
+inserts the filter vf=crop=w:h:x:y, where w,h,x,y are determined
+from the vf-metadata gathered by cropdetect. The cropdetect filter
+is removed immediately after the crop filter is inserted as it is
+no longer needed.
+
+If the "C" key is pressed again, the crop filter is removed
+restoring playback to its original state.
+
+Since the crop parameters are determined from the 1 second of video
+between inserting the cropdetect and crop filters, the "C" key
+should be pressed at a position in the video where the crop region
+is unambiguous (i.e., not a black frame, black background title
+card, or dark scene).
+
+The default options can be overridden by adding
+script-opts-append=autocrop-<parameter>=<value>
+into mpv.conf
+
+List of available parameters (For default values, see <local options>)：
+
+auto: bool - Whether to automatically apply crop at the start of playback.
+    If you don't want crop automatically, set it to false or
+    add "script-opts-append=autocrop-auto=no" into mpv.conf.
+
+auto_delay: seconds - Delay before starting crop.
+    this only effective when playback time is less than this.
```

Comment:
"The auto mode is only effective. . ." Also this sentence should be after the one below.

Reply:
**The auto mode is only effective when playback time is less than this.**
Seems doesn't right.

**The auto delay is only effective when playback time is less than this.**
Should it be this?

Reply:
No, it's not the delay that is ineffective. Too large a delay causes whole automatic mode of operation to be ineffective.